src/GitHub.TeamFoundation.14/Services/TeamExplorerServices.cs

```diff
@@ -110,5 +123,37 @@ void OpenFolder(string repositoryPath)
             var dte = serviceProvider.TryGetService<DTE>();
             dte?.ExecuteCommand("File.OpenFolder", repositoryPath);
         }
+
+        static void BringIntoView(ITeamExplorerSection section)
+        {
+            var control = section?.SectionContent as UserControl;
+            if (control != null)
+            {
+                if (control.IsLoaded)
+                {
+                    BringIntoView();
+                }
+                else
+                {
+                    control.Loaded += (s, e) => BringIntoView();
+                }
+            }
+
+            void BringIntoView()
+            {
+                var targetRectangle = new Rect(0, 0, 0, 1000);
+                control.BringIntoView(targetRectangle);
+            }
```

Comment:
Not sure if this is the best idea, but sometimes I create custom awaitables:

```cs
var control = section?.SectionContent as UserControl;
if (control is null) return;

await control.WhenLoaded(); // No-op if IsLoaded is true, continues inside the Loaded event handler if not
var targetRectangle = new Rect(0, 0, 0, 1000);
control.BringIntoView(targetRectangle);
```

Reply:
Good idea. Actually, I could do something like this using Rx. Could you sanity check my implementation?

Reply:
I haven't used Rx, but everything about the code you have makes sense to me.